### PR TITLE
Feature/graceful stop

### PR DESCRIPTION
**Pull request by lfmundim**
## Issue:
I sometimes am crawling but I need to stop and nothing was found so far, so the bot is still running. If I click "Disconnect" the app gets angry at me throwing an exception. 
image

## Solution:
To mitigate that I added a "Stop" button (with a tooltip) that will immediately stop the advancing loop. The same behavior is now applied to the "Disconnect" button as well.

I had to move the UI around a bit to accommodate the new button

image

Issue 2 of #114

### CODE REVIEW

**Comment by jnovack** [comment]
Rather than move the UI around, merge the locations of the "Connect/Disconnect" buttons, and put stop in Disconnect's location.

Connect/Disconnect are never active at the same time, they are wasting space.

File: MainWindow.cs [hunk 1]
```diff
@@ -898,14 +909,27 @@ private async void ButtonAdvanceDate_Click(object sender, EventArgs e)
...
                 ButtonAdvanceDate.Enabled = false;
+                ButtonStopAdvance.Enabled = true;
                 _WindowState = WindowState;
                 var prompt = false;
+
                 do
                 {
+                    // average run takes between 15-18secs on a v1 switch, undocked
+                    // if a single run of the loop takes more than a minute, stop it, something is wrong
+                    AdvanceDateCancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(60));
```
Comment (jnovack) [inline MainWindow.cs:920]: I love this.

**Comment by lfmundim** [comment]
> Rather than move the UI around, merge the locations of the "Connect/Disconnect" buttons, and put stop in Disconnect's location.
> 
> Connect/Disconnect are never active at the same time, they are wasting space.

that would leave Advance/Stop separate, from a UX POV that is not very intuitive though is it?

**Comment by lfmundim** [comment]
Any chance this is being considered by the owner? If so I'll resolve the conflict

**Comment by LegoFigure11** [comment]
Superseded by #137

**Pull request closed without merging**